implementation/README.md

```diff
@@ -0,0 +1,147 @@
+The aim of LinuxBoot is to reduce complexity and obscure firmware by moving that functionality into kernel and userspace. This chapter describes the procedures from a [LinuxBoot workshop](https://docs.google.com/presentation/d/1s9ka4v7leKeJa3116AQoNb9cv3OqmnW6pgn0ov9WiHo/edit?ts=5e2b227b#slide=id.g7ceec54197_4_163) where an Atomic Pi board with UEFI firmware was converted to run LinuxBoot. 
+
+## A quick refresher on UEFI
+
+UEFI has three sections:
+
++   SEC ("Boot") 
++   PEI ("Very early chip setup and DRAM programming")
++   DXE ("DRAM code")
+
+The first two are about 1/16 of the total image. DXE is the rest. It contains drivers that might include:
+
++   Bugs
++   Sendmail programs
++   Web servers
++   Exploits (computrace and other phone home software)
++   Uploaders
+
+DXE process is also very complex; some systems have 750 DXEs.
+LinuxBoot replaces most of the UEFI software with Linux. LinuxBoot has an initramfs provided by [u-root](../u-root/README.md).
+
+## How do you get LinuxBoot on your hardware?
+
+Start with a board running standard UEFI and proceed from "zero changes to FLASH" to "max changes" in 4 steps:
+
+1.  Boot from USB stick via UEFI shell command _or_ netboot (zero changes)
+1.  Replace UEFI Shell code section with Linux kernel (change part of one thing)
+1.  Remove as many DXEs as possible (change by removal). This change:
+
+       *  Speeds boot
+       *  Reduces panic possibilities
+       *  Removes exploits
+       *  In production, it has solved problems
+
+1.  Replace some DXEs with open source components (change by replacement)
+
+## Tools of the trade
+
+There are two tools you use when you modify the UEFI flash image: `utk `and `me_cleaner`
+
+The `utk` tool can:
+
++   Remove DXEs
++   Insert new DXEs
++   Replace the binary code of a DXE with a kernel
+
+If you are lucky, you can always flash with _internal_ flash hardware.
+
+The Management Engine (ME) is an x86 CPU embedded in the Intel Platform Controller Hub (PCH). It runs the Minix operating system which boots first and enables hardware such as clocks and GPIOs. ME checks the contents of flash memory and is used to implement "BootGuard". If you reflash and the ME is in "BootGuard" mode, your machine will be unusable. You need to run a tool called `me_cleaner` on the image to disable BootGuard.
+
+     /usr/bin/python2 me_cleaner.py -s somefile.bin
+
+`me_cleaner` sets the high assurance platform (HAP) bit. HAP is an IT security framework that provides a way to disable a feature on Intel chips that presents a security risk. Setting the bit with `me_cleaner` disables the "feature".  Note that this does not always work; check with your hardware vendor.  
```

Comment:
"presents a security risk" -> "does not allow us to modify the UEFI image and install LinuxBoot"

Again, this is true, based on the many CVEs against the ME,  but ... maybe we don't need to say it.

As for "check with your hardware vendor", hard experience says an answer from the vendors is as useful as a coin flip. 

"check with the linuxboot community."

Reply:
I made these changes.